spec/syntax.md

```diff
@@ -619,31 +619,34 @@ quoted-char    = %x0-5B         ; omit \
                / %xE000-10FFFF
 
 unquoted       = unquoted-start *name-char
-unquoted-start = name-start / DIGIT / "."
-               / %xB7 / %x300-36F / %x203F-2040
+unquoted-start = ALPHA / DIGIT / "." / "_"
+               / %xB7 / %xC0-D6 / %xD8-F6 / %xF8-37D
+               / %x37F-1FFF / %x200C-200D / %x203F-2040
+               / %x2070-218F / %x2C00-2FEF / %x3001-D7FF
+               / %xF900-FDCF / %xFDF0-FFFD / %x10000-EFFFF
 ```
 
 ### Names
 
 A **_<dfn>name</dfn>_** is an identifier for a _variable_ (prefixed with `$`),
 for a _function_ (prefixed with `:`, `+` or `-`),
 or for an _option_ (these have no prefix).
-The namespace for _names_ is based on XML's [Name](https://www.w3.org/TR/xml/#NT-Name),
-with the restriction that it MUST NOT start with `:`,
-as that would conflict with the _function_ start character.
-Otherwise, the set of characters allowed in names is large.
+The namespace for _name_ matches XML's [Name](https://www.w3.org/TR/xml/#NT-Name).
+
+As `:` is also used as the start sigil of _function_,
+using a _name_ with it as a first character is NOT RECOMMENDED.
```

Comment:
I disagree with this. We have namespacing in another PR just now and there's a reasonable solution: instead of XML `Name` use XML-Name's `NCName` as the basis. The definition of `NCName` is exactly "`Name` minus the `:` character"

Reply:
Why should there be such an unnecessarily complex overlap between variable/function/option names and unquoted literals? If the function sigil were just replaced with something that does not appear in [XML `NameChar`](https://www.w3.org/TR/xml/#NT-NameChar) then the names could be exactly described by [XML `Name`](https://www.w3.org/TR/xml/#NT-Name) and the unquoted literals by either `1*name-char` (i.e., [XML `Nmtoken`](https://www.w3.org/TR/xml/#NT-Nmtoken)) or by `1*(name-char / "+" / …)`, in either case forming a strict superset rather than a _near_-superset to exclude artificially-induced ambiguity w.r.t. colons.